news/_posts/2024-06-03-spark-4.0.0-preview1.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: post
+title: Preview release of Spark 4.0
+categories:
+- News
+tags: []
+status: publish
+type: post
+published: true
+meta:
+  _edit_last: '4'
+  _wpas_done_all: '1'
+---
+To enable wide-scale community testing of the upcoming Spark 4.0 release, the Apache Spark community has posted a preview release of
+Spark 4.0. This preview is not a stable release in terms of either API or functionality, but it is meant to give the community early
+access to try the code that will become Spark 4.0. If you would like to test the release, please download it, and send feedback using
+either the mailing lists or JIRA.
+
+There are a lot of exciting new features added to Spark 4.0, including ANSI mode by default, Python data source, polymorphic
+Python UDTF, string collation support, new VARIANT data type, streaming state store data source, structured logging, Java 17 by default,
+and many more.
+
+We'd like to thank our contributors and users for their contributions and early feedback to this release. This release would not have
+been possible without you.
+
+To download Spark 4.0.0-preview1, head over to the <a href="https://archive.apache.org/dist/spark/spark-4.0.0-preview1">download page</a>.
```

Comment:
Is it deliberate to skip the docker release for 4.0.0-preview1?

Reply:
In the last preview release, there was no docker release, so there is no precedent we can follow. I'm not sure if we need to do it or not, cc @dongjoon-hyun @Yikun 

It looks like just some manual steps in the spark-docker repo and docker hub, shouldn't be hard.

Reply:
IMO, this step shouldn't be skipped and shall be added to the release guide if absent.